Sources/TuistKit/Services/InitService.swift

```diff
@@ -136,6 +170,36 @@ class InitService {
         }
     }
 
+    /// Extract branch's name if it exist from given template URL. If not return nil
+    ///
+    ///  - Parameter templateURL: given template URL
+    private func parseRepoBranch(from templateURL: String) -> String? {
```

Comment:
You will need this method and `parseRepoURL` for `scaffold`, too. Maybe we could move these to `GitHandler`? I'd also make sure to add proper unit tests showcasing the functions' functionality as you did for `isGitURL`

Reply:
I'd instead create a `TemplateLocationParser` utility. `GitHandler` is a wrapper around local Git operations.